lib/internal/modules/cjs/loader.js

```diff
@@ -306,21 +306,30 @@ function readPackage(requestPath) {
   }
 }
 
+/**
+ * @param {string} checkPath
+ * @returns {!({
+ *   data: PjsonObject,
+ *   path: string,
+ * } | false)}
+ */
 function readPackageScope(checkPath) {
   const rootSeparatorIndex = checkPath.indexOf(path.sep);
-  let separatorIndex;
-  while (
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
```

Comment:
Does the whole fix not just work by turning this `>` into a `>=`?

Reply:
```diff
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
+   (separatorIndex = checkPath.lastIndexOf(path.sep)) >= rootSeparatorIndex
```

Do you mean if the only change was the above line? Nope, that alone doesn't do the trick!